src/test/java/seedu/address/logic/commands/AddCommandTest.java

```diff
@@ -1,195 +1,194 @@
-//         public void setRecipe(Recipe target, Recipe editedRecipe) {
-//             throw new AssertionError("This method should not be called.");
-//         }
-//
-//         @Override
-//         public ObservableList<Recipe> getFilteredRecipeList() {
-//             throw new AssertionError("This method should not be called.");
-//         }
-//
-//         @Override
-//         public void updateFilteredRecipeList(Predicate<Recipe> predicate) {
-//             throw new AssertionError("This method should not be called.");
-//         }
-//     }
-//
-//     /**
-//      * A Model stub that contains a single person.
-//      */
-//     private class ModelStubWithRecipe extends ModelStub {
-//         private final Recipe recipe;
-//
-//         ModelStubWithRecipe(Recipe recipe) {
-//             requireNonNull(recipe);
-//             this.recipe = recipe;
-//         }
-//
-//         @Override
-//         public boolean hasRecipe(Recipe recipe) {
-//             requireNonNull(recipe);
-//             return this.recipe.isSameRecipe(recipe);
-//         }
-//     }
-//
-//     /**
-//      * A Model stub that always accept the recipe being added.
-//      */
-//     private class ModelStubAcceptingRecipeAdded extends ModelStub {
-//         final ArrayList<Recipe> recipesAdded = new ArrayList<>();
-//
-//         @Override
-//         public boolean hasRecipe(Recipe recipe) {
-//             requireNonNull(recipe);
-//             return recipesAdded.stream().anyMatch(recipe::isSameRecipe);
-//         }
-//
-//         @Override
-//         public void addRecipe(Recipe recipe) {
-//             requireNonNull(recipe);
-//             recipesAdded.add(recipe);
-//         }
-//
-//         @Override
-//         public ReadOnlyRecipeBook getRecipeBook() {
-//             return new RecipeBook();
-//         }
-//     }
-//
-// }
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.function.Predicate;
+
+import org.junit.jupiter.api.Test;
+
+import javafx.collections.ObservableList;
+import seedu.address.commons.core.GuiSettings;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.ReadOnlyRecipeBook;
+import seedu.address.model.ReadOnlyUserPrefs;
+import seedu.address.model.RecipeBook;
+import seedu.address.model.recipe.Recipe;
+import seedu.address.testutil.RecipeBuilder;
+
+public class AddCommandTest {
+
+    @Test
+    public void constructor_nullRecipe_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new AddCommand(null));
+    }
+
+    @Test
```

Comment:
I think testing for execute should utilise the `CommandTestUtil#assertCommandFailure` or `CommandTestUtil#assertCommandSuccess` method.

This is an example of how it was done in DeleteCommand:
```java
@Test
public void execute_validIndexFilteredList_success() {
    showRecipeAtIndex(model, INDEX_FIRST_RECIPE);

    Recipe recipeToDelete = model.getFilteredRecipeList().get(INDEX_FIRST_RECIPE.getZeroBased());
    DeleteCommand deleteCommand = new DeleteCommand(INDEX_FIRST_RECIPE);

    String expectedMessage = String.format(DeleteCommand.MESSAGE_DELETE_RECIPE_SUCCESS, recipeToDelete);

    Model expectedModel = new ModelManager(model.getRecipeBook(), new UserPrefs());
    expectedModel.deleteRecipe(recipeToDelete);
    showNoRecipe(expectedModel);

    assertCommandSuccess(deleteCommand, model, expectedMessage, expectedModel);
}
```

Reply:
Refactored one of the tests to use assertCommandSuccess, however the assertThrows tests can't be replaced with assertCommandFailure since it uses the getRecipeBook() method which is disabled in the ModelStub class